src/commands/createNewProject/CSharpProjectCreator.ts

```diff
@@ -36,9 +38,23 @@ export class CSharpProjectCreator extends ProjectCreatorBase {
         this._runtime = await tryGetLocalRuntimeVersion() || await promptForProjectRuntime(this.ui);
         const identity: string = `Microsoft.AzureFunctions.ProjectTemplate.CSharp.${this._runtime === ProjectRuntime.one ? '1' : '2'}.x`;
         await executeDotnetTemplateCommand(this._runtime, this.functionAppPath, 'create', '--identity', identity, '--arg:name', projectName);
+
+        if (!this._detectedRuntime) {
+            await this.detectRuntime();
```

Comment:
Do we need to call this here and on line 49?

Reply:
Yes that's actually the main fix for the bug. There's several different entry points for C# projects and not all of them were calling `getRuntime`. I refactored most of the logic into the separate `detectRuntime` function so that I could call it twice